mmv1/products/gkeonprem/VmwareCluster.yaml

```diff
@@ -0,0 +1,662 @@
+      The object name of the VMware OnPremUserCluster custom resource on the
+      associated admin cluster. This field is used to support conflicting
+      names when enrolling existing clusters to the API. When used as a part of
+      cluster enrollment, this field will differ from the ID in the resource
+      name. For new clusters, this field will match the user provided cluster ID
+      and be visible in the last component of the resource name. It is not
+      modifiable.
+
+      All users should use this name to access their cluster using gkectl or
+      kubectl and should expect to see the local name when viewing admin
+      cluster controller logs.
+    output: true
+  - !ruby/object:Api::Type::String
+    name: "etag"
+    description: |
+      This checksum is computed by the server based on the value of other
+      fields, and may be sent on update and delete requests to ensure the
+      client has an up-to-date value before proceeding.
+      Allows clients to perform consistent read-modify-writes
+      through optimistic concurrency control.
+    output: true
+  - !ruby/object:Api::Type::KeyValuePairs
+    name: "annotations"
+    description: |
+      Annotations on the VMware User Cluster.
+      This field has the same restrictions as Kubernetes annotations.
+      The total size of all keys and values combined is limited to 256k.
+      Key can have 2 segments: prefix (optional) and name (required),
+      separated by a slash (/).
+      Prefix must be a DNS subdomain.
+      Name must be 63 characters or less, begin and end with alphanumerics,
+      with dashes (-), underscores (_), dots (.), and alphanumerics between.
+    default_from_api: true
+  - !ruby/object:Api::Type::NestedObject
+    name: "controlPlaneNode"
+    description: |
+      VMware User Cluster control plane nodes must have either 1 or 3 replicas.
+    required: true
+    properties:
+      - !ruby/object:Api::Type::Integer
+        name: cpus
+        description: |
+          The number of CPUs for each admin cluster node that serve as control planes
+          for this VMware User Cluster. (default: 4 CPUs)
+        default_from_api: true
+      - !ruby/object:Api::Type::Integer
+        name: memory
+        description: |
+          The megabytes of memory for each admin cluster node that serves as a
+          control plane for this VMware User Cluster (default: 8192 MB memory).
+        default_from_api: true
+      - !ruby/object:Api::Type::Integer
+        name: replicas
+        description: |
+          The number of control plane nodes for this VMware User Cluster.
+          (default: 1 replica).
+        default_from_api: true
+      - !ruby/object:Api::Type::NestedObject
+        name: autoResizeConfig
+        description: |
+          AutoResizeConfig provides auto resizing configurations.
+        default_from_api: true
+        properties:
+          - !ruby/object:Api::Type::Boolean
+            name: enabled
```

Comment:
I think this can be set to required? Same w/ other nested objects holding a single boolean.

Reply:
Added. Thanks for the suggestion